src/web/app/pages-session/session-submission-page/session-submission-page.component.spec.ts

```diff
@@ -30,17 +578,520 @@ describe('SessionSubmissionPageComponent', () => {
+      feedbackResponseCommentId: 999,
+      commentText: 'comment text',
+      createdAt: 10000000,
+      lastEditedAt: 10000000,
+      isVisibilityFollowingFeedbackQuestion: true,
+      showGiverNameTo: [CommentVisibilityType.GIVER, CommentVisibilityType.INSTRUCTORS],
+      showCommentTo: [CommentVisibilityType.GIVER, CommentVisibilityType.INSTRUCTORS],
+    };
+    const testSubmissionForm: FeedbackResponseRecipientSubmissionFormModel = deepCopy(testMcqRecipientSubmissionForm);
+    const expectedId: any = testMcqRecipientSubmissionForm.commentByGiver?.originalComment?.feedbackResponseCommentId;
+    const commentSpy: Spy = spyOn(feedbackResponseCommentService, 'updateComment').and.returnValue(of(testComment));
+
+    component.createCommentRequest(testSubmissionForm).subscribe(() => {
+      expect(testSubmissionForm.commentByGiver).toEqual(component.getCommentModel(testComment));
+    });
+
+    expect(commentSpy.calls.count()).toEqual(1);
+    expect(commentSpy.calls.mostRecent().args[0]).toEqual({
+      commentText: 'comment text here',
+      showCommentTo: [],
+      showGiverNameTo: [],
+    });
+    expect(commentSpy.calls.mostRecent().args[1]).toEqual(expectedId);
+    expect(commentSpy.calls.mostRecent().args[2]).toEqual(Intent.STUDENT_SUBMISSION);
+    expect(commentSpy.calls.mostRecent().args[3]).toEqual({ key: testQueryParams.key, moderatedperson: '' });
+  });
+
+  it('should create comment request to delete existing comment when text is empty', () => {
+    const testSubmissionForm: FeedbackResponseRecipientSubmissionFormModel = deepCopy(testMcqRecipientSubmissionForm2);
+    const expectedId: any = testMcqRecipientSubmissionForm2.commentByGiver?.originalComment?.feedbackResponseCommentId;
+    const commentSpy: Spy = spyOn(feedbackResponseCommentService, 'deleteComment').and.returnValue(of({}));
+
+    component.createCommentRequest(testSubmissionForm).subscribe(() => {
+      expect(testSubmissionForm.commentByGiver).toEqual(undefined);
+    });
+
+    expect(commentSpy.calls.count()).toEqual(1);
+    expect(commentSpy.calls.mostRecent().args[0]).toEqual(expectedId);
+    expect(commentSpy.calls.mostRecent().args[1]).toEqual(Intent.STUDENT_SUBMISSION);
+    expect(commentSpy.calls.mostRecent().args[2]).toEqual({ key: testQueryParams.key, moderatedperson: '' });
+  });
+
+  it('should delete participant comment', () => {
+    const testSubmissionForm: QuestionSubmissionFormModel = deepCopy(testMcqQuestionSubmissionForm);
+    const expectedId: any = testMcqQuestionSubmissionForm.recipientSubmissionForms[0]
+        .commentByGiver?.originalComment?.feedbackResponseCommentId;
+    const commentSpy: Spy = spyOn(feedbackResponseCommentService, 'deleteComment').and.returnValue(of(true));
+
+    component.questionSubmissionForms = [testSubmissionForm];
+    component.deleteParticipantComment(0, 0);
+
+    expect(commentSpy.calls.count()).toEqual(1);
+    expect(commentSpy.calls.mostRecent().args[0]).toEqual(expectedId);
```

Comment:
maybe some comments explaining what is `commentSpy.calls.mostRecent().args[0]` can make the test clearer